sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts

```diff
@@ -301,12 +318,235 @@ export class IER12ProcessingService {
         totalAssessedCost: assessmentData.totalAssessedCost,
         totalAssessmentNeed: assessmentData.totalAssessmentNeed,
         disbursementSentDate: disbursement.dateSent,
+        applicationEventCode: await this.getApplicationEventCode(
+          pendingAssessment,
+          disbursement,
+        ),
       };
       ier12Records.push(ier12Record);
     }
     return ier12Records;
   }
 
+  /**
+   * Get application event code (i.e current state of an application)
+   * @param assessment assessment of institutions.
+   * @param currentDisbursementSchedule current disbursement schedule.
+   * @returns application event code.
+   */
+  private async getApplicationEventCode(
+    assessment: StudentAssessment,
+    currentDisbursementSchedule: DisbursementSchedule,
+  ): Promise<ApplicationEventCode> {
+    const application = assessment.application;
+    switch (application.applicationStatus) {
+      case ApplicationStatus.Assessment:
+        return this.applicationEventCodeDuringAssessment(
+          application.applicationNumber,
+        );
+
+      case ApplicationStatus.Enrolment:
+        return this.applicationEventCodeDuringEnrolmentAndCompleted(
+          currentDisbursementSchedule.coeStatus,
+        );
+
+      case ApplicationStatus.Completed:
+        return this.applicationEventCodeDuringCompleted(
+          currentDisbursementSchedule,
+          application.student.id,
+        );
+
+      case ApplicationStatus.Cancelled:
+        return ApplicationEventCode.DISC;
+    }
+  }
+
+  /**
+   * Get application event code for an application with assessment status.
+   * @param applicationNumber application number.
+   * @returns application event code.
+   */
+  private async applicationEventCodeDuringAssessment(
+    applicationNumber: string,
+  ): Promise<ApplicationEventCode.REIA | ApplicationEventCode.ASMT> {
+    // Check if the application has more than one submissions.
+    switch (
+      await this.applicationService.hasMultipleApplicationSubmissions(
+        applicationNumber,
+      )
+    ) {
+      case true:
+        return ApplicationEventCode.REIA;
+      case false:
+        return ApplicationEventCode.ASMT;
+    }
+  }
+
+  /**
+   * Get application event code for an application with enrollment/completed status.
+   * @param coeStatus coe status.
+   * @returns application event code.
+   */
+  private async applicationEventCodeDuringEnrolmentAndCompleted(
```

Comment:
same here 

https://github.com/bcgov/SIMS/pull/2374/files#r1353142336

Reply:
Here the check is done with only the COE required and declined status, do you still prefer a conditional operator here? This method is re-used in 2 places, one place needs COE approval and the other doesn't.
![image](https://github.com/bcgov/SIMS/assets/77353155/47844c88-899f-4ce3-b2ae-7989c755e124)
![image](https://github.com/bcgov/SIMS/assets/77353155/ef343877-c7dc-47ec-8e93-150ba06e04c6)
![image](https://github.com/bcgov/SIMS/assets/77353155/8e44d0d1-5778-4d50-aff8-dab28b4a317a)
IMO, for this case SWITCH is cleaner than a conditional statement. Please let me know your thought

Reply:
I got your point, and I am good with switch. I would just recommend adding a throw error as default. We can ask for dev opinion on the same. @andrewsignori-aot @andrepestana-aot @guru-aot  ?

Reply:
I wouldn`t mind having a throwing error on `switch default`.